src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -54,5 +67,36 @@ public void arcadeDrive(double staightSpeed, double turnModifer) {
       public void arcadeDrive(DrivingDeltas drivingDeltas) {
         arcadeDrive(drivingDeltas.getForwardPower(), drivingDeltas.getSteeringPower());
       }
+
+      public void resetEncoders(){
+        leftEncoder.reset();
+        rightEncoder.reset();
+      }
+ 
+
+      public void driveStraight(double distance, double maxSpeed){
+        System.out.println("B Button Pressed");
+        leftGearbox.setBrakeMode();
+        rightGearbox.setCoastMode();
+        resetEncoders();
+        while(leftEncoder.getDistance() < distance || rightEncoder.getDistance() < distance) {
+       // System.out.println("Left Distance Traveled" + leftEncoder.getDistance()); 
+       // System.out.println("Right Distance Traveled" + rightEncoder.getDistance());
+        double error = leftEncoder.getDistance() - rightEncoder.getDistance();
+        double turnPower =  error * .15;
+        double fowardSpeed = maxSpeed - ((leftEncoder.getDistance()/distance)*maxSpeed);
+        System.out.println("forward speed: " + fowardSpeed);
+       //double turnPower = 0;
+         arcadeDrive(-fowardSpeed, -turnPower);
```

Comment:
this needs to have some formatting love, and no commented out lines please! Lets get rid of the prints too, if you're done debugging.